packages/provider/src/multi-wallet.ts

```diff
@@ -0,0 +1,271 @@
+import { Signer as AbstractSigner, Contract, ethers } from 'ethers'
+import { TransactionResponse, JsonRpcProvider } from 'ethers/providers'
+import { Wallet } from './wallet'
+import { NetworkConfig, ArcadeumWalletConfig, ArcadeumContext, Transactionish } from './types'
+import { Arrayish, BigNumberish } from 'ethers/utils'
+import { abi as mainModuleUpgradableAbi } from './abi/mainModuleUpgradable'
+import { abi as requireUtilsAbi } from './abi/requireUtils'
+import { isConfig } from './utils'
+import { NotEnoughSigners } from './errors'
+
+// TODO: Add more details to network
+// authChain, mainNetwork, etc
+export type NetworkWallet = {
+  wallet: Wallet,
+  network: NetworkConfig
+}
+
+export type MultiWalletParams = {
+  context: ArcadeumContext,
+  initialConfig: ArcadeumWalletConfig,
+  signers: AbstractSigner[],
+  networks: NetworkConfig[]
+}
+
+type FullConfig = {
+  threshold: Threshold[],
+  signers: Signer[]
+}
+
+type Threshold = {
+  chaind: number,
+  weight: number
+}
+
+type Signer = {
+  address: string,
+  networks: {
+    chaind: number,
+    weight: number
+  }[]
+}
+
+export class MultiWallet extends AbstractSigner {
+  private readonly _wallets: NetworkWallet[]
+  provider: ethers.providers.JsonRpcProvider
+
+  constructor(params: MultiWalletParams) {
+    super()
+
+    if (params.networks.length === 0) throw new Error('MultiWallet must have networks')
+
+    // Generate wallets using the initial configuration
+    this._wallets = params.networks.map((network) => {
+      let wallet = new Wallet(params.initialConfig, params.context, ...params.signers)
+      wallet.setProvider(network.rpcUrl)
+      if (network.relayer) {
+        wallet.setRelayer(network.relayer)
+      }
+      return {
+        network: network,
+        wallet: wallet
+      }
+    })
+
+    this.provider = this._wallets[0].wallet.provider
+  }
+
+  get address(): string {
+    return this._wallets[0].wallet.address
+  }
+
+  getAddress(): Promise<string> {
+    return this._wallets[0].wallet.getAddress()
+  }
+
+  async getFullConfig(): Promise<FullConfig> {
```

Comment:
what is a "full config" ? versus just a normal wallet config?

Reply:
Because there are multiple networks, the wallet may have different configurations on each one of those networks. This method returns all the configuration of all the current networks.